client/checkout/woopay/direct-checkout/woopay-direct-checkout.js

```diff
@@ -0,0 +1,167 @@
+			hiddenDiv.style.visibility = 'hidden';
+			hiddenDiv.style.position = 'absolute';
+			hiddenDiv.style.height = '0';
+
+			checkoutButton.parentElement.appendChild( hiddenDiv );
+
+			ReactDOM.render(
+				<WooPayConnectIframe
+					listeners={ this.listeners }
+					actionCallback={ this.actionCallback }
+				/>,
+				hiddenDiv
+			);
+		}
+	}
+
+	static async setTempThirdPartyCookie() {
+		const setTempThirdPartyCookie = new Promise( ( resolve ) => {
+			this.listeners.setTempThirdPartyCookie = resolve;
+		} );
+
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( { action: 'setTempThirdPartyCookie' } );
+
+		// Since iFrame is refreshed, we need to re-init the iframePostMessage promise.
+		this.state.iframePostMessage = new Promise( ( resolve ) => {
+			this.listeners.setIframePostMessage = resolve;
+		} );
+
+		return await setTempThirdPartyCookie;
+	}
+
+	static async isWooPayThirdPartyCookiesEnabled() {
+		const isCookieSet = await this.setTempThirdPartyCookie();
+		if ( ! isCookieSet ) {
+			return false;
+		}
+
+		const isWooPayThirdPartyCookiesEnabled = new Promise( ( resolve ) => {
+			this.listeners.setIsWooPayThirdPartyCookiesEnabled = resolve;
+		} );
+
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( { action: 'getIsThirdPartyCookiesEnabled' } );
+
+		return await isWooPayThirdPartyCookiesEnabled;
+	}
+
+	static async isUserLoggedIn() {
+		const isUserLoggedIn = new Promise( ( resolve ) => {
+			this.listeners.setIsUserLoggedIn = resolve;
+		} );
+
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( { action: 'getIsUserLoggedIn' } );
+
+		return await isUserLoggedIn;
+	}
+
+	static async sendSessionDataToWooPay() {
+		const woopaySessionData = new Promise( ( resolve ) => {
+			this.listeners.setWoopaySessionResponse = resolve;
+		} );
+
+		const woopaySession = await this.getWooPaySessionFromMerchant();
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( {
+			action: 'setRedirectSessionData',
+			value: woopaySession,
+		} );
+
+		const { redirect_url: redirectUrl } = await woopaySessionData;
+
+		return redirectUrl;
+	}
+
+	static getCheckoutRedirectElements() {
+		// TODO: Add other checkout elements, if needed.
+		const checkoutButton = document.querySelector(
+			'.wc-proceed-to-checkout .checkout-button'
+		);
+
+		return [ checkoutButton ];
+	}
+
+	static redirectToWooPaySession( elements ) {
+		elements.forEach( ( element ) => {
+			element.addEventListener( 'click', async ( event ) => {
+				event.preventDefault();
+
+				const woopayRedirectUrl = await this.sendSessionDataToWooPay();
+
+				window.location.href = woopayRedirectUrl;
+			} );
+		} );
+	}
+
+	static redirectToWooPay( elements ) {
+		elements.forEach( ( element ) => {
+			element.addEventListener( 'click', async ( event ) => {
+				event.preventDefault();
+
+				const woopayRedirectUrl = await this.sendSessionDataToWooPay();
```

Comment:
This might be a pre-emptive optimization, but I was wondering if this redirect URL could be stored once on the object (or maybe even on a merchant level cookie?) so that we don't have to create a new WooPay session each time? I've confused myself a bit on this topic already so just wanted to ask the question. Same idea could apply to `redirectToWooPaySession`.

Reply:
There are a lot of options to go with here, but that discussion fits better under our two optimizations task under milestone 2 in the https://github.com/Automattic/woopay/issues/2440.